clang/bindings/python/clang/cindex.py

```diff
@@ -200,13 +236,16 @@ class _CXString(Structure):
 
     _fields_ = [("spelling", c_char_p), ("free", c_int)]
 
-    def __del__(self):
+    def __del__(self) -> None:
         conf.lib.clang_disposeString(self)
 
     @staticmethod
-    def from_result(res, fn=None, args=None):
+    def from_result(res: _CXString, fn: Any = None, args: Any = None) -> str:
         assert isinstance(res, _CXString)
-        return conf.lib.clang_getCString(res)
+        pystr: str | None = conf.lib.clang_getCString(res)
+        if pystr is None:
+            return ""
+        return pystr
```

Comment:
`conf.lib.clang_getCString` may sometimes (though seemingly rarely) return `None` instead of a Python `str`. `_CXString.from_result()` is used in a lot of places throughout this file, and while parts of it seem to be aware of this (e.g. doc string saying that this may return `None`) other places are not (e.g. calling `len` on the return value). Many parts of the interface also directly return the result of this function, and having to check for `None` in all of these places seems potentially impractical, so I went for returning empty strings instead of `None` instead.

But due to the inconsistent usage throughout this file, I'm not sure how far this corresponds more or less to the original intention.

Reply:
This change makes sense. I'm not sure it's breaking, though, because nothing breaks for both users who have been checking for `None`, and those who have not been.

Reply:
The interface changes, isn't that a breaking change in every case? I.e. if you were checking for empty results before by doing `is None`, that doesn't work anymore. You would now have to change that to `== ""`.
I've added a breaking change note for this, but happy to remove this again if I'm missing something here.